src/main/kotlin/org/rust/cargo/project/configurable/RustfmtConfigurable.kt

```diff
@@ -5,14 +5,71 @@
 
 package org.rust.cargo.project.configurable
 
+import com.intellij.execution.configuration.EnvironmentVariablesComponent
+import com.intellij.openapi.options.BoundConfigurable
+import com.intellij.openapi.options.ConfigurationException
 import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.ComboBox
 import com.intellij.openapi.ui.DialogPanel
+import com.intellij.ui.RawCommandLineEditor
+import com.intellij.ui.components.Label
 import com.intellij.ui.layout.panel
+import com.intellij.ui.layout.toBinding
 import org.rust.RsBundle
+import org.rust.cargo.project.settings.rustfmtSettings
+import org.rust.cargo.toolchain.RustChannel
+
+class RustfmtConfigurable(private val project: Project) : BoundConfigurable(RsBundle.message("settings.rust.rustfmt.name")) {
+    private val state = project.rustfmtSettings.settingsState
```

Comment:
I think we don't need to hold copy of the state here. Just write to the settings directly. It will also makes custom `apply` redundant

Reply:
But what if I want to undo the changes? The `Cancel` button won't work if I write the settings directly.

Reply:
It will work since you provide `modelBinding`. Could you provide an implementation that doesn't work for you?